Remove Slack and update `CONTRIBUTING`

tnull opened the pull request:
Since it came up on Discord, I now also removed Slack and updated `CONTRIBUTING`.

tnull commented:
@jkczyz Is your LDK Dev Meet calendar so official that you'd be comfortable having it added to `CONTRIBUTING.md` or the meeting notes page?

jkczyz commented:
> @jkczyz Is your LDK Dev Meet calendar so official/public that you'd be comfortable having it added to `CONTRIBUTING.md` or the meeting notes page?

Yeah, in the sense that I update it if a meeting is canceled.

ariard commented on CONTRIBUTING.md:
```diff
@@ -1,53 +1,59 @@
+Anyone is invited to contribute without regard to technical experience,
+"expertise", OSS experience, age, or other concern. However, the development of
+cryptocurrencies demands a high-level of rigor, adversarial thinking, thorough
+testing and risk-minimization. Any bug may cost users real money. That being
```
"risk-minimization engineering process"

tnull replied:
Hm, I don't think adding this checks out? Would read "However, the development of cryptocurrencies demands a high-level of (...) risk-minimization engineering process". So I think the current form is more correct?

ariard replied:
Sure, just risk-minimization doesn't seem to refer what is actually qualified risk-minimized.

ariard commented on CONTRIBUTING.md:
```diff
@@ -1,53 +1,59 @@
+cryptocurrencies demands a high-level of rigor, adversarial thinking, thorough
+testing and risk-minimization. Any bug may cost users real money. That being
+said, we deeply welcome people contributing for the first time to an open source
+project or pick up Rust while contributing. Don't be shy, you'll learn.
```
maybe add ref to #rust-101 on Discord

tnull replied:
Good idea, added in 6136b32.

ariard reviewed (commented):
Good

tnull commented on CONTRIBUTING.md:
```diff
@@ -1,53 +1,59 @@
-development compared to your merged work.
+This doesn't mean don't be ambitious with the breadth and depth of your
+contributions but rather understand the project culture before investing an
+asymmetric number of hours on development compared to your merged work.
```
While I see the motivation, I'm hesitant to put such a upfront defensive warning here under 'Getting Started'. Can we maybe continue to handle this on an individual basis if it comes up?

ariard replied:
Sure, we'll do our best.

tnull commented:
> Yeah, in the sense that I update it if a meeting is canceled.

Great, I now added a calendar link. In fact, I added the embedded version, since the other required me to log in to Google.

tnull commented on CONTRIBUTING.md:
```diff
@@ -1,53 +1,64 @@
 
-Communication about Rust-Lightning happens primarily on #ldk-dev on the
-[LDK slack](http://www.lightningdevkit.org/), but also #rust-bitcoin on IRC Freenode.
+Communication about the devevelopment of LDK and `rust-lightning` happens
```
I believe 'Communications' is also correct and it's just a question of writing style?

tnull replied:
Done.

tnull commented on CONTRIBUTING.md:
```diff
@@ -1,53 +1,64 @@
-[LDK slack](http://www.lightningdevkit.org/), but also #rust-bitcoin on IRC Freenode.
+Communication about the devevelopment of LDK and `rust-lightning` happens
+primarily on the [LDK Discord](https://discord.gg/5AcknnMfBw) in the `#ldk-dev`
+channel. Additionally, live LDK devevelopment meetings are held every other
```
lol. Done.

tnull commented:
> I assume you can go ahead and squash.

Squashed!
